### DISCUSSION TITLE
GFPGAN keep skin texture?

### DISCUSSION THREAD

**Discussion by Telecino333** [hlky/stable-diffusion-webui#1265] [Q&A]
Is there any way to alter GFPGAN to correct the eyes, but keep the skin texture as is?

Or, to reduce the amount of correction overall?

**Comment by dudumaroja** [comment]
you could save both images and photoshop in the changes you want.  i often only upscale and gan images that i like a lot.. this way you can finetune the adjustments

**Reply by AnaTool** [reply]
i think the GFPGAN model is the problem. V1.3 is much more smoother than V1.0. An option to change the models will be cool. V1 can also do bring color back into the pictures.

https://github.com/TencentARC/GFPGAN/blob/master/inference_gfpgan.py

https://github.com/TencentARC/GFPGAN/blob/master/Comparisons.md

# ------------------------ set up GFPGAN restorer ------------------------
    if args.version == '1':
        arch = 'original'
        channel_multiplier = 1
        model_name = 'GFPGANv1'
        url = 'https://github.com/TencentARC/GFPGAN/releases/download/v0.1.0/GFPGANv1.pth'
    elif args.version == '1.2':
        arch = 'clean'
        channel_multiplier = 2
        model_name = 'GFPGANCleanv1-NoCE-C2'
        url = 'https://github.com/TencentARC/GFPGAN/releases/download/v0.2.0/GFPGANCleanv1-NoCE-C2.pth'
    elif args.version == '1.3':
        arch = 'clean'
        channel_multiplier = 2
        model_name = 'GFPGANv1.3'
        url = 'https://github.com/TencentARC/GFPGAN/releases/download/v1.3.0/GFPGANv1.3.pth'
    elif args.version == '1.4':
        arch = 'clean'
        channel_multiplier = 2
        model_name = 'GFPGANv1.4'
        url = 'https://github.com/TencentARC/GFPGAN/releases/download/v1.3.0/GFPGANv1.4.pth'
    elif args.version == 'RestoreFormer':
        arch = 'RestoreFormer'
        channel_multiplier = 2
        model_name = 'RestoreFormer'
        url = 'https://github.com/TencentARC/GFPGAN/releases/download/v1.3.4/RestoreFormer.pth'
    else:
        raise ValueError(f'Wrong model version {args.version}.')
